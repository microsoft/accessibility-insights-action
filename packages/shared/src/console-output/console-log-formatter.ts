// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { toolName } from '../content/strings';

export const link = (href: string, text: string): string => {
    return `${text} (${href})`;
};

export const listItem = (text: string): string => {
    return `* ${text}`;
};

export const productTitle = (): string => {
    return `${toolName}`;
};

export const footerSeparator = (): string => `-------------------`;

export const sectionSeparator = (): string => '\n';
