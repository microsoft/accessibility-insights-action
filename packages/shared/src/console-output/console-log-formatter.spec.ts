// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { footerSeparator, link, listItem, productTitle, sectionSeparator } from './console-log-formatter';

describe('ConsoleLogFormatter', () => {
    it('link', () => {
        expect(link('href', 'text')).toMatchSnapshot();
    });

    it('listItem', () => {
        expect(listItem('li')).toMatchSnapshot();
    });

    it('productTitle', () => {
        expect(productTitle()).toMatchSnapshot();
    });

    it('footerSeparator', () => {
        expect(footerSeparator()).toMatchSnapshot();
    });

    it('sectionSeparator', () => {
        expect(sectionSeparator()).toMatchSnapshot();
    });
});
