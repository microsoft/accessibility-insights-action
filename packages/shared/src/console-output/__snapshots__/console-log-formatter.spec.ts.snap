// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsoleLogFormatter footerSeparator 1`] = `"-------------------"`;

exports[`ConsoleLogFormatter link 1`] = `"text (href)"`;

exports[`ConsoleLogFormatter listItem 1`] = `"* li"`;

exports[`ConsoleLogFormatter productTitle 1`] = `"Accessibility Insights"`;

exports[`ConsoleLogFormatter sectionSeparator 1`] = `
"
"
`;
